### PR TITLE
Add grants data for patents

### LINKED ISSUE #146: Add patents granted metric
Add data regarding the number of applications that a company has filed for AI patents.

parat-patent-applications-statbox

### PR DESCRIPTION
**Pull request by rggelles**
Pipeline runs in airflow, so hopefully all is well.

Closes #146

### CODE REVIEW

File: company_linkage/sql/linked_ai_patents_grants.sql [hunk 1]
```diff
@@ -0,0 +1,121 @@
+-- Pulling every AI-associated patent family id linked to every grid id of any assignee for that patent, and all the assignee names
```
Comment (jmelot) [inline company_linkage/sql/linked_ai_patents_grants.sql:1]: ROR id now, right?

File: company_linkage/sql/linked_ai_patents_grants.sql [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+-- We also pull in the AI subcategories and the years
+-- We also attempt to add in "fake" families for the patents that are missing patent families
+with patents_orig as (
+SELECT
+  -- Pulling in the current assignee ror ids from dimensions
```
Comment (jmelot) [inline company_linkage/sql/linked_ai_patents_grants.sql:6]: This comment needs an update

**Review by jmelot — COMMENTED** [review]
I happened to notice a couple of comments need to be updated but otherwise don't see issues with this. Thank you! I'll keep this PR open until I've integrated your changes into the data preprocessing script, will ask Brian to review that and integrate into the UI when it's done.

**Comment by rggelles** [comment]
I tried to rebase here and I don't know if I did it right, so...maybe check that before you merge? It got weirdly mad at my attempt.

**Comment by jmelot** [comment]
@brianlove can you please review https://github.com/georgetown-cset/parat/pull/196/commits/a8c05058ff0b935b1b4e2b32fa2d7c88d8a2f77f ? I already reviewed Rebecca's changes, but that commit contains my changes to integrate the new metrics (I also made two later commits to be less overzealous in updating the requirements.txt and to fix the tests...)

**Review by brianlove — APPROVED** [review]
👍

**Pull request merged**